### PR TITLE
nattraversal linting

### PR DESCRIPTION
**Pull request by noamnelke**
## Motivation
Linting errors in the `nat_traversal` package.

## Changes
Rename the package from `nat_traversal` to `nattraversal`.
Add comments.

### CODE REVIEW

File: p2p/swarm_test.go [hunk 1]
```diff
@@ -1370,8 +1370,8 @@ func (u *UpnpGatewayMock) Clear(port uint16) error {
 	return nil
 }
 
-func createDiscoverUpnpFunc(funcErr error, port uint16, gatewayForwardErr error) func() (igd nat_traversal.UpnpGateway, err error) {
-	return func() (igd nat_traversal.UpnpGateway, err error) {
+func createDiscoverUpnpFunc(funcErr error, port uint16, gatewayForwardErr error) func() (igd nattraversal.UpnpGateway, err error) {
+	return func() (igd nattraversal.UpnpGateway, err error) {
```
Comment (y0sher) [inline p2p/swarm_test.go]: some places we use `Upnp` and some places we use `UPnP` (`discoverUPnPGateway`). lets try to be consistent ? I know using a lot of capital letters might not look the best. but usually acronyms should be written with capital letters..

**Review by y0sher — APPROVED** [review]

File: p2p/swarm_test.go (hunk 1 above)
Comment (noamnelke) [reply p2p/swarm_test.go]: Good point. Fixed this.

**Comment by noamnelke** [comment]
bors merge

**Comment by noamnelke** [comment]
Merged as part of **Consolidated linting (#1958)**

**Pull request closed without merging**